### PR TITLE
Always collect core web vitals data for user in variant group of AB test

### PR DESCRIPTION
**Pull request by zekehuntergreen**
## What does this change?
Before the change, we collect core web vitals data for 1% of users.
After the change we always collect the data if the user is in the variant group of a server-side AB test, otherwise fall back to 1% sampling.

## Does this change need to be reproduced in dotcom-rendering ?
We are accepting the fact that after this change, the behavior around collection of core web vitals in frontend and DCR will be different.
The equivalent change would need to be made [here](https://github.com/guardian/dotcom-rendering/blob/dc6ab694db8a9b52285e0d30552e120f98012b67/src/web/browser/coreVitals/coreVitals.ts#L31) but isn't possible because server-side AB test groups aren't exposed to DCR. 

## Screenshots
N/A

## What is the value of this and can you measure success?
Often we need to determine whether or not the groups in an AB test have an impact on core web vitals. When the AB test is run on a very small group, say a 1% of all users, the current sampling rate of 1% means that gathering an appropriate volume of data takes a very long time. This change solves the problem by ignoring the sampling rate for users participating in a test.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/analytics/coreVitals.ts [hunk 1]
```diff
@@ -30,8 +30,16 @@ const jsonData: CoreWebVitalsPayload = {
...
-	const inSample = Math.floor(Math.random() * 100);
-	if (inSample != 1) {
+	// If the current user is part of a server-side AB test
+	// then we always want to sample their core web vitals data.
+	const userInTest =
+		window.guardian.config.tests &&
+		Object.values(window.guardian.config.tests).includes('variant');
+
+	// Otherwise, only send core web vitals data for 1% of users.
+	const inSample = Math.floor(Math.random() * 100) == 1;
```
Comment (zekehuntergreen) [inline static/src/javascripts/projects/common/modules/analytics/coreVitals.ts:40]: To me this definition of `inSample` makes the variable's name more accurate.

Comment (mxdvl) [reply static/src/javascripts/projects/common/modules/analytics/coreVitals.ts:40]: I would suggest the following, which allows to easily edit the right-hand-side to any other percentage value:

```suggestion
	const inSample = Math.random() < 1 / 100;
```

File: static/src/javascripts/types/global.d.ts [hunk 2]
```diff
@@ -65,6 +65,10 @@ interface CommercialPageConfig {
 interface Config {
 	page: PageConfig;
 	switches: Record<string, boolean | undefined>;
+	tests?: {
+		testName: string;
+		groupName: string;
+	};
```
Comment (mxdvl) [inline static/src/javascripts/types/global.d.ts:71]: I don’t this this type is accurate. Based on sample data, `window.guardian.config.tests
` looks more like this:

```ts
{
    interactiveLibrarianVariant: "variant", // from the live website
    someTestControl: "control", // IIRC
    anotherTestVariant: "variant", // IIRC
}
```

Therefore, the type would be closer to:

```suggestion
	tests?: Record<string, 'control' | 'variant'>;
```

**Review by mxdvl — CHANGES_REQUESTED** [review]
Thanks for this!

I think that the `coreVitals.ts` should explicitly link to [dotcom-
rendering]’s one and I don’t thikn the type of `guardian.config.tests` is quite right.

File: static/src/javascripts/types/global.d.ts (hunk 2 above)
Comment (zekehuntergreen) [reply static/src/javascripts/types/global.d.ts:71]: Yes, good call!

**Review by mxdvl — APPROVED** [review]

**Pull request merged**